biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -230,7 +270,10 @@ public PutResult put(InputStream stream, PutOptions options) throws Exception {
 	@Override
 	public File get(String bsn, Version version, Map<String, String> properties, DownloadListener... listeners)
 		throws Exception {
-		init();
+		prepare();
```

Comment:
don't use prepare. You can have an init() but prepare should catch all errors & is only called once.

Reply:
Hmm... If I have a prepare that simply checks the config and sets a status if something is off, I would need to check the `isOk()` in the init because it would most likely throw an Exception. The I would need to check everywhere the `isOk()` as well (as I already do), to avoid Exceptions e.g. in BndTools. Would this be right?

Reply:
The prepare method is from the Prepare interface, it has a well defined meaning, so it should not be called as the check method. In prepare() you calculate _anything_ that is related to the properties since they will not change during the life of the instance. Since you do not want to block, ideally you want to do any initialization in a background thread. An init() method could then block until that thread is finished & check isOk().

     prepare() -> process properties, registry, etc. & start init thread
     init()          -> block till init thread is ready, check status